### PR TITLE
[Feature] Support list partition names from iceberg

### PR DESCRIPTION
**Pull request by Youngwb**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

#11116 

## Problem Summary(Required) ：

Support list identity partition names from iceberg

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by mofeiatwork** [comment]
run starrocks_clang-format

**Comment by wangruin** [comment]
run starrocks_clang-format

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 12 / 31 (38.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergMetadata.java|0|7|00.00%|[96, 97, 98, 99, 102, 103, 107]|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|12|24|50.00%|[195, 196, 197, 198, 199, 202, 203, 206, 207, 208, 209, 210]|

**Review by murphyatwork — APPROVED** [review]

**Review by mxdzs0612 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by Youngwb** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java [hunk 1]
```diff
@@ -89,4 +90,20 @@ public Table getTable(String dbName, String tblName) {
...
+
+    @Override
+    public List<String> listPartitionNames(String dbName, String tblName) {
+        org.apache.iceberg.Table icebergTable
+                = icebergCatalog.loadTable(IcebergUtil.getIcebergTableIdentifier(dbName, tblName));
+        if (!IcebergCatalogType.fromString(catalogType).equals(IcebergCatalogType.HIVE_CATALOG)) {
+            throw new StarRocksIcebergException(
+                    "Do not support get partitions from catalog type: " + catalogType);
+        }
+        if (icebergTable.spec().fields().stream().anyMatch(partitionField -> !partitionField.transform().isIdentity())) {
```
[the remark this answers is not in the document]
Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java:102]: I think we can support non-identity later, I have not figured out how to distinguish identity and non-identity from Partition name which is string type

**Review by gengjun-git — APPROVED** [review]

**Comment by Youngwb** [comment]
run starrocks_admit_test

**Pull request merged**